tutorials/haskell/snap-web-app/tutorial.md

```diff
@@ -0,0 +1,480 @@
+---
+title: How to create a web application in Haskell with Snap
+published: 2016-05-24
+ghc: 7.10.3
+lts: 5.17
+libraries: snap-core-0.9.8.0 snap-extras-0.11.0.2 snap-server-0.9.5.1 heist-0.14.1.4 HDBC-sqlite3-2.3.3.1    
+language: haskell
+author-name: Omar García
+---
+
+## Introduction
+
+[Snap](http://snapframework.com/) is a simple and [fast](http://snapframework.com/blog/2010/11/17/snap-0.3-benchmarks)
+web development framework for the Haskell programming language with a clean
+[monads](https://wiki.haskell.org/Monad) oriented design. Snap apps are often
+composed by [Snaplets](http://snapframework.com/docs/tutorials/snaplets-tutorial),
+auto-contained pieces of reusable functionalities that can be glued together to build a website.
+
+In this tutorial we'll show you how to build a simple TODO web app using Snap,
+[SQLite](https://www.sqlite.org/), and [Stack](http://docs.haskellstack.org/en/stable/README/)
+as our Haskell build tool. The target app should deliver the folllowing features:
+
+ - List Todo items
+ - Create a new Todo item
+ - Mark a Todo item as Done
+ - Delete a Todo item
+
+
+## Getting Started
+
+Snap comes shipped with a [scaffolding utility](http://snapframework.com/docs/quickstart)
+to scrap some boilerplate. So, when you first run the `snap init` command into an empty
+directory, it will create a bunch of files for you. But lets leave that aside,
+for now lets focus on how to create an Snap web app from scratch using Stack.
+
+Firstly, lets [create a project using Stack](http://docs.haskellstack.org/en/stable/README/#start-your-new-project) by running the following command:
+
+`stack new todo-webapp`
```

Comment:
@nebtrx This will probably look better by using a block:

    ```
    stack new todo-webapp
    ```

Instead of:

    `stack new todo-webapp`

There's another "lets" here to be changed.